hyperspy/models/edsmodel.py

```diff
@@ -0,0 +1,822 @@
+    if weight_line is None:
+        weight_line = elements_db[
+            element]['Atomic_properties']['Xray_lines'][line]['weight']
+    return lambda x: x * weight_line
+
+
+def _get_iweight(element, line, weight_line=None):
+    if weight_line is None:
+        weight_line = elements_db[
+            element]['Atomic_properties']['Xray_lines'][line]['weight']
+    return lambda x: x / weight_line
+
+
+def _get_sigma(E, E_ref, units_factor):
+    # 2.5 from Goldstein, / 1000 eV->keV, / 2.355^2 for FWHM -> sigma
+    return lambda sig_ref: math.sqrt(abs(
+        4.5077 * 1e-4 * (E - E_ref) * units_factor + np.power(sig_ref, 2)))
+
+
+def _get_offset(E, diff):
+    return lambda E: E + diff
+
+
+def _get_scale(E1, E_ref1, fact):
+    return lambda E: E1 + fact * (E - E_ref1)
+
+
+class EDSModel(Model1D):
+    """Build a fit a model for EDS instance
+
+    Parameters
+    ----------
+    spectrum : an EDSSpectrum (or any EDSSpectrum subclass) instance
+    auto_add_lines : boolean
+        If True, automatically add Gaussians for all X-rays generated
+        in the energy range by an element, using the edsmodel.add_family_lines
+        method
+    auto_background : boolean
+        If True, adds automatically a polynomial order 6 to the model,
+        using the edsmodel.add_polynomial_background method.
+
+    Example
+    -------
+    >>> m = create_model(s)
+    >>> m.fit()
+    >>> m.fit_background()
+    >>> m.calibrate_energy_axis('resolution')
+    >>> m.calibrate_xray_lines('energy',['Au_Ma'])
+    >>> m.calibrate_xray_lines('sub_weight',['Mn_La'],bound=10)
+    """
+
+    def __init__(self, spectrum,
+                 auto_add_lines=True,
+                 *args, **kwargs):
+        Model1D.__init__(self, spectrum, *args, **kwargs)
+        self.xray_lines = list()
+        self.background_components = list()
+        end_energy = self.axes_manager.signal_axes[0].high_value
+        if self.spectrum._get_beam_energy() < end_energy:
+            self.end_energy = self.spectrum._get_beam_energy()
+        else:
+            self.end_energy = end_energy
+        self.start_energy = self.axes_manager.signal_axes[0].low_value
+        units_name = self.axes_manager.signal_axes[0].units
+        if units_name == 'eV':
+            self.units_factor = 1000.
+        elif units_name == 'keV':
+            self.units_factor = 1.
+        else:
+            raise ValueError("Energy units, %s, not supported" %
+                             str(units_name))
+        if auto_add_lines is True:
+            self.add_family_lines()
+
+    @property
+    def spectrum(self):
+        return self._spectrum
+
+    @spectrum.setter
+    def spectrum(self, value):
+        if isinstance(value, EDSSpectrum):
+            self._spectrum = value
+        else:
+            raise ValueError(
+                "This attribute can only contain an EDSSpectrum "
+                "but an object of type %s was provided" %
+                str(type(value)))
```

Comment:
Since it EDSModel inherits from Model1D I think spectrum is correct here... Essentially the structure is that anything in BaseModel is .signal anything in Model1D (and by extension here) is .spectrum and anything in Model2D is .image .... the __init__ for Model1D then days self.signal = self.spectrum for 1D and vice-versa. Does that make sense?